EmpireLegends/GameEngine.cpp

```diff
@@ -14,9 +14,11 @@ GameEngine::GameEngine()
 
 GameEngine::~GameEngine()
 {
-	delete gameStart;
-	delete startPhase;
-	delete mainGame;
+	gameStart = nullptr;
```

Comment:
No, you need to make the map `nullptr` in each class's destructor AFTER deleting it. Here you're not even deleting the objects.